packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts

```diff
@@ -420,6 +435,329 @@ export function mikroOrmBaseRepositoryFactory<T extends object = object>(
+        relation.reference === ReferenceType.MANY_TO_ONE
+      ) {
+        // If it is undefined (not passed) then just return
+        if (entryCopy[relation.name] === undefined) {
+          return
+        }
+
+        // If null is passed (to be unset) then we need to set that on the join column
+        if (entryCopy[relation.name] === null) {
+          delete entryCopy[relation.name]
+          entryCopy[relation.joinColumns[0]] = null
+          return
+        }
+
+        const newEntity = this.getEntityWithId(
+          manager,
+          relation.type,
+          entryCopy[relation.name]
+        )
+
+        delete entryCopy[relation.name]
+        entryCopy[relation.joinColumns[0]] = newEntity.id
+
+        if (!upsertsPerType[relation.type]) {
+          upsertsPerType[relation.type] = []
+        }
+        upsertsPerType[relation.type].push(newEntity)
+
+        return newEntity
+      }
+
+      if (allRelationsToHandle.includes(relation.name)) {
+        delete entryCopy[relation.name]
+      }
+
+      return
+    }
+
+    // Returns a POJO object with the ID populated from the entity model hooks
+    protected getEntityWithId(
+      manager: SqlEntityManager,
+      entityName: string,
+      data: any
+    ): Record<string, any> & { id: string } {
+      const created = manager.create(entityName, data, {
+        managed: false,
+        persist: false,
+      })
+
+      return { id: (created as any).id, ...data }
+    }
+
+    protected async upsertManyToOneRelations_(
+      manager: SqlEntityManager,
+      upsertsPerType: Record<string, any[]>
+    ) {
+      const typesToUpsert = Object.entries(upsertsPerType)
+      if (!typesToUpsert.length) {
+        return []
+      }
+
+      return (
+        await promiseAll(
+          typesToUpsert.map(async ([type, data]) => {
+            return await this.upsertMany_(manager, type, data)
+          })
+        )
+      ).flat()
+    }
+
+    protected async upsertMany_(
+      manager: SqlEntityManager,
+      entityName: string,
+      entries: any[]
+    ) {
+      const existingEntities: any[] = await manager.find(
+        entityName,
+        {
+          id: { $in: entries.map((d) => d.id) },
+        },
+        {
+          populate: [],
+          disableIdentityMap: true,
+        }
+      )
+      const existingEntitiesMap = new Map(
+        existingEntities.map((e) => [e.id, e])
+      )
+
+      const createdEntities: T[] = []
+      const updatedEntities: T[] = []
+      const orderedEntities: T[] = []
+
+      await promiseAll(
+        entries.map(async (data) => {
+          orderedEntities.push(data)
+          const existingEntity = existingEntitiesMap.get(data.id)
+          if (existingEntity) {
+            await manager.nativeUpdate(entityName, { id: data.id }, data)
+            updatedEntities.push(data)
+          } else {
+            await manager.insert(entityName, data)
+            createdEntities.push(data)
```

Comment:
we need to keep it as we will need to return what have been created/deleted/updated later for the event management

Reply:
@riqwan we could yeah, it was left there for when we fire off events

Reply:
suggestion: We could remove the `ordering` from the upsert method and let that be someone, you'd have to do separately. Then, `upsertMany` could have a return type of `{ created, updated }`.

Alternatively, just do `{ created, updated, ordered }`.

Reply:
@olivermrbl I would probably return {created, updated, ordered} rather than just created and updated as there are places where the ordering does matter.